devtools/src/main/scala/kalix/devtools/impl/KalixProxyContainer.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.devtools.impl
+
+import com.typesafe.config.Config
+import kalix.devtools.BuildInfo
+import kalix.devtools.impl.KalixProxyContainer.KalixProxyContainerConfig
+import org.slf4j.LoggerFactory
+import org.testcontainers.Testcontainers
+import org.testcontainers.containers.GenericContainer
+import org.testcontainers.containers.output.Slf4jLogConsumer
+import org.testcontainers.utility.DockerImageName
+
+object KalixProxyContainer {
+
+  object KalixProxyContainerConfig {
+    def apply(config: Config): KalixProxyContainerConfig = {
+
+      // FIXME: core Kalix should be able to read dev-mode.conf from root folder as well
+      // as such, we can kalix.user-function-port configured to something else without touching the prod config
```

Comment:
Still evaluating if `ROOT/dev-model.conf` is better then `src/main/resources/dev-model.conf` + exclusion.